contracts/src/v0.8/vrf/dev/SubscriptionAPI.sol

```diff
@@ -405,14 +409,18 @@ abstract contract SubscriptionAPI is ConfirmedOwner, IERC677Receiver, IVRFSubscr
     if (consumers.length == MAX_CONSUMERS) {
       revert TooManyConsumers();
     }
-    mapping(uint256 => uint64) storage nonces = s_consumers[consumer];
-    if (nonces[subId] != 0) {
+    mapping(uint256 => ConsumerConfig) storage consumerConfigs = s_consumers[consumer];
+    ConsumerConfig consumerConfig = consumerConfigs[subId];
+    if (consumerConfig.active) {
       // Idempotence - do nothing if already added.
       // Ensures uniqueness in s_subscriptions[subId].consumers.
       return;
     }
-    // Initialize the nonce to 1, indicating the consumer is allocated.
-    nonces[subId] = 1;
+    // consumerConfig.nonce is 0 if the consumer had never sent a request to this subscription
+    // otherwise, consumerConfig.nonce is non-zero
+    // in both cases, use consumerConfig.nonce as is and set active status to true
+    consumerConfig.active = true;
+    consumerConfigs[subId] = consumerConfig;
```

Comment:
Does `consumerConfigs[subId].active = true` work? Think need to set the above to `storage`. Need to test to see which way is cheaper.

Reply:
looks like this is the cheapest:

```
ConsumerConfig storage consumerConfig = s_consumers[consumer][subId];
    if (consumerConfig.active) {
      // Idempotence - do nothing if already added.
      // Ensures uniqueness in s_subscriptions[subId].consumers.
      return;
    }
    // consumerConfig.nonce is 0 if the consumer had never sent a request to this subscription
    // otherwise, consumerConfig.nonce is non-zero
    // in both cases, use consumerConfig.nonce as is and set active status to true
    consumerConfig.active = true;
```